app/models/activity.rb

```diff
@@ -1,135 +1,32 @@
-class Activity < ApplicationRecord
-  include InputSanitizeHelper
-
-  # after_create :generate_notification
+# frozen_string_literal: true
 
-  enum type_of: [
-    :create_project,
-    :rename_project,
-    :change_project_visibility,
-    :archive_project,
-    :restore_project,
-    :assign_user_to_project,
-    :change_user_role_on_project,
-    :unassign_user_from_project,
-    :create_module,
-    :clone_module,
-    :archive_module,
-    :restore_module,
-    :change_module_description,
-    :assign_user_to_module,
-    :unassign_user_from_module,
-    :create_step,
-    :destroy_step,
-    :add_comment_to_step,
-    :complete_step,
-    :uncomplete_step,
-    :check_step_checklist_item,
-    :uncheck_step_checklist_item,
-    :edit_step,
-    :add_result,
-    :add_comment_to_result,
-    :archive_result,
-    :edit_result,
-    :create_experiment,
-    :edit_experiment,
-    :archive_experiment,
-    :clone_experiment,
-    :move_experiment,
-    :add_comment_to_project,
-    :edit_project_comment,
-    :delete_project_comment,
-    :add_comment_to_module,
-    :edit_module_comment,
-    :delete_module_comment,
-    :edit_step_comment,
-    :delete_step_comment,
-    :edit_result_comment,
-    :delete_result_comment,
-    :destroy_result,
-    :start_edit_wopi_file,
-    :unlock_wopi_file,
-    :load_protocol_from_file,
-    :load_protocol_from_repository,
-    :revert_protocol,
-    :create_report,
-    :delete_report,
-    :edit_report,
-    :assign_sample,
-    :unassign_sample,
-    :complete_task,
-    :uncomplete_task,
-    :assign_repository_record,
-    :unassign_repository_record
-  ]
-
-  validates :type_of, presence: true
+class Activity < ApplicationRecord
+  serialize :values, Hash
+  enum type_of: Extends::ACTIVITY_TYPES
 
-  belongs_to :project, inverse_of: :activities
+  belongs_to :owner, inverse_of: :activities, class_name: 'User'
+  belongs_to :subject, polymorphic: true, optional: true
+  belongs_to :project, inverse_of: :activities # for permissions
```

Comment:
@okriuchykhin  
Correct me if I am wrong. 
In the current implementation of activities, `project` behaves sometimes as a `subject` and sometimes is there just for permissions check? Is that true? 

So are we going to store `project` from now on in `values` (when is a part of action but not main) or in `subject`(when is a main "actor") AND in `project` association always for permission check?
Please correct me if I am wrong.  
cc @radiokills

Reply:
The `project` is optional, the `team` is required always. Both are for permissions check.